### ISSUE TITLE
Checkin action no longer stores comments in Alfresco's versionDescription property

### ISSUE THREAD

**Issue by dgrumieaux** [tsgrp/HPI#1976]
With the new updates to pass in ad hoc properties to the CheckinActionExecuter, the version comment is no longer passed into the IContent#checkin method. This is the only place we currently have to save the Alfresco versionDescription field, so with the current implementation ViewVersions will not be able to show comments for newly checked in documents.

The desired fix will allow the versionDescription property to be set upon checkin, not to use a separate property as Alfresco's versionDescription property is specific to the cm:workingcopy aspect.

**Comment by gsteimer** [comment]
While doing this, we also need to revisit how the checkin action executer was changed.  The updates make it so there's a checkin as well as an update object.  This will cause two audit entries vs. the expected one.

Before starting this, let's meet with @dgrumieaux and @benallenallen to come up with the best approach for all repos

**Comment by gsteimer** [comment]
Items we talked through

- Question about using checkin / updateObject vs. enhancedUpdateObject
- Need a slider in the checkin action - "include legacy comments field (Alfresco only)"
  - if on, include in the interface (in the form, or below?)
  - if on, include the comments in the call to enhancedUpdateObject as part of the property map with a "cheater" key
- In Alfresco enhancedUpdateObject, if versioning the object (`newVersion == true`):
  - check for the "cheater key in the map"
  - if it's there, pull out the value and remove the key from the map
  - do Alfresco magic before the checkin line ~1035 to set the comments on the working copy

**Comment by dhruvvora94** [comment]
Closed with OC commit #23763 and OCMS commit #24102

**Comment by gsteimer** [comment]
Seeing some issues on edge 1 and edge2

Edge2:
With a form configured and "Alfresco comments" mode on the comments box appears but the form does not (Configured on Controlled Docs):
![image](https://user-images.githubusercontent.com/1377228/68426377-9f115e00-016d-11ea-8b1f-6a6635271329.png)

Edge1:
With a form configured and "Alfresco comments" mode OFF, the legacy comments box appears anyway:
![image](https://user-images.githubusercontent.com/1377228/68426437-c405d100-016d-11ea-9212-75fc46b75f94.png)

Admin:
In the admin, there's no option to _not_ configure a form.  I'd assume that we'd want to NOT configure a form in a lot of cases on alfresco.
![image](https://user-images.githubusercontent.com/1377228/68426559-ffa09b00-016d-11ea-91c9-955c8cf3a971.png)

**Comment by dhruvvora94** [comment]
The issued mentioned here were due to configuration errors. In Alfresco. controlled docs are a parent type. We will need to configure a form on all the child types in order for it to work the way we expect it to. 

Comments do not work on Controlled documents and this has been documented in a separate issue https://github.com/tsgrp/HPI/issues/1991
